frontend/src/App.svelte

```diff
@@ -275,80 +277,146 @@ const displayInfoMessage= () =>{ //After 1 minute of no nodes recieved, a messag
           enableOptimize: true,
         },
       ],
+    },
+  };
+
+  const handle_close_request = () => {
+    console.log("received in app");
+    show_settings = !show_settings;
+  };
+
+  const handleKeyDown = (e: KeyboardEvent): void => {
+    appKeyMap[e.key] = e.type == "keydown";
+
+    if (
+      (appKeyMap["ArrowRight"] ||
+        appKeyMap["ArrowLeft"] ||
+        appKeyMap["ArrowUp"] ||
+        appKeyMap["ArrowDown"]) &&
+      nonInteractiveState
+    ) {
+      nonInteractiveState = false;
+    }
+
+    // Hide all panels
+    if (appKeyMap["h"] || appKeyMap["H"]) {
+      if (show_legend) toggleLegend();
+
+      if (show_settings) toggleSettings();
+
+      if (show_filter_panel) toggleFilterPanel();
+
+      if (show_timebar) updateTimebar();
+    }
+
+    // Legend panel
+    if (appKeyMap["l"] || appKeyMap["L"]) {
+      toggleLegend();
+    }
+
+    // Options panel
+    if (appKeyMap["s"] || appKeyMap["S"]) {
+      toggleSettings();
+    }
+
+    // Non-interactive mode
+    if (appKeyMap["n"] || appKeyMap["N"]) {
+      toggleInteractiveMode();
+    }
+
+    // Timebar
+    if (appKeyMap["t"] || appKeyMap["T"]) {
+      updateTimebar();
+    }
+
+    // Filter panel (waiting for implementation of toggle filter panel)
+    if (appKeyMap["f"] || appKeyMap["F"]) {
+      toggleFilterPanel();
+    }
+
+    // WIP: Shortcut view
+    if (appKeyMap["i"] || appKeyMap["I"]) {
+      console.log("WIP: Shortcut view");
     }
   };
 
- const  handle_close_request = () => {
-    console.log('received in app')
-    show_settings = !show_settings
- }
+  const handleKeyUp = (e: KeyboardEvent): void => {
```

Comment:
I am not sure how this works, why do we need to handle key up?

Reply:
We handle keyup because it registers when a button is no longer pressed.

If we only handled keydown we would end up with O = true, LeftArrow = true, RightArrow = true, etc. By also listening for the key's release we can update the keyMap and swap the boolean states of the individual keys.

Reply:
Ok thanks for clarification :D